packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -641,26 +635,12 @@ export class SchemaFactory<
 		const Name extends TName,
 		const T extends Unenforced<ImplicitAllowedTypes>,
 	>(name: Name, allowedTypes: T) {
-		class RecursiveArray extends this.namedArray(
+		const RecursiveArray = this.namedArray(
 			name,
 			allowedTypes as T & ImplicitAllowedTypes,
 			true,
 			false,
-		) {
```

Comment:
I take it the constructor logic here is already handled by `namedArray`?

Reply:
There is no logic here, so yes. Note that the constructor just contains an if where both branches are identical invocations of super(data)